docs/src/examples/Supervised-NN-ODE-MNIST.md

```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
+otherwise use Array( x )" ? </mark>
+
+### Load MNIST Dataset into Minibatches
+
+The preprocessing is done in `loadmnist` where the raw MNIST data is split into features `x_train`
+and labels `y_train` by specifying batchsize `bs`. The function `convertlabel` will then transform
+the current labels (`labels_raw`) from numbers 0 to 9 (`LabelEnc.NativeLabels(collect(0:9))`) into
+one hot encoding (`LabelEnc.OneOfK`).
+
+Features are reshaped into format <b>[Height, Width, Color, BatchSize]</b> or in this case <b>[28, 28, 1, 128]</b>
+meaning that every minibatch will contain 128 images with a single color channel of 28x28 pixels.
+The entire dataset of 60,000 images is then split into these batches with `batchview` resulting in
+468 different batches (60,000 / 128) for both features and labels.
+
+```julia
+function loadmnist(batchsize = bs)
+	#Use MLDataUtils LabelEnc for natural onehot conversion
+  	onehot(labels_raw) = convertlabel(LabelEnc.OneOfK, labels_raw, LabelEnc.NativeLabels(collect(0:9)))
+	#Load MNIST
+	imgs, labels_raw = MNIST.traindata();
+	#Process images into (H,W,C,BS) batches
+	x_train = Float32.(reshape(imgs,size(imgs,1),size(imgs,2),1,size(imgs,3))) |> gpu
+	x_train = batchview(x_train,batchsize)
+	#Onehot and batch the labels
+	y_train = onehot(labels_raw) |> gpu
+	y_train = batchview(y_train,batchsize)
+	return x_train, y_train
+end
+```
+
+And then loaded from main
+```
+# Main
+const bs = 128
+x_train, y_train = loadmnist(bs)
+```
+
+
+### Layers
+
+The Neural Network requires passing inputs sequentially through multiple layers.
+`Chain` allows inputs to functions to come from the previous layer and send the outputs
+to the next. Four different sets of layers are used here.
+
+
+```julia
+down = Chain(x->reshape(x,(28*28,:)),
+             Dense(784,20,tanh)
+            ) |> gpu
+
+nn  = Chain(Dense(20,10,tanh),
+            Dense(10,10,tanh),
+            Dense(10,20,tanh)
+           ) |> gpu
+
+
+nn_ode = NeuralODE(nn, (0.f0, 1.f0), Tsit5(),
+                        save_everystep = false,
+                        reltol = 1e-3, abstol = 1e-3,
+                        save_start = false) |> gpu
+
+fc  = Chain(Dense(20,10)) |> gpu
+
+nfe = 0
+```
+
+`down`: Takes a 28x28 image, flattens it, then passes it through a fully connected
+layer with `tanh` activation<br>
+`nn`: Neural network chain three layers deep, `tanh` activation <br>
+`nn_ode`: ODE solver for neural network <mark><-- Should add more explanation </mark><br>
+`fc`: Final fully connected layer <br>
+`nfe`: <mark><-- not sure </mark><br>
+`|> gpu`: sends to GPU
+
+### Array Conversion
+
+Cheap conversion of `DiffEqArray` from ODE solver into Matrix.
+
+```julia
+function DiffEqArray_to_Array(x)
+    xarr = gpu(x)
+    return reshape(xarr, size(xarr)[1:2])
+end
+```
+
+This is required to convert the output of the Neural Network ODE.
+
+### Build Topology
+
+Connect all layers together in a single chain
```

Comment:
I think it would be good if this has a narrative flow. "Next we connect all of the layers together ..."